src/main/scala/com/fulcrumgenomics/vcf/api/Variant.scala

```diff
@@ -100,7 +100,7 @@ final case class Variant(chrom: String,
                          attrs: ListMap[String,Any] = Variant.EmptyInfo,
                          genotypes: Map[String, Genotype] = Variant.EmptyGenotypes
                         ) {
-  require(genotypes.values.forall(_.alleles == alleles), "Genotypes must have the same alleles as their Variant.")
+  require(gts.forall(_.alleles == alleles), "Genotypes must have the same alleles as their Variant.")
```

Comment:
How about:
```suggestion
  require(gts.forall(gt => gt.alleles eq alleles || gt.alleles == alleles), "Genotypes must have the same alleles as their Variant.")
```

Reply:
I worry a bit about parsing e.g. large VCFs with thousands of samples/genotypes and the overhead.  But when we read/construct from the Java versions, we construct the allele set once and re-use it, so checking for identity first and then equality should short-circuit fairly effectively.